Cubical/Functions/Involution.agda

```diff
@@ -0,0 +1,43 @@
+{-# OPTIONS --cubical --no-import-sorts --safe #-}
+
+module Cubical.Functions.Involution where
+
+open import Cubical.Foundations.Equiv
+open import Cubical.Foundations.Isomorphism
+open import Cubical.Foundations.Prelude
+open import Cubical.Foundations.Univalence
+
+isInvolution : ∀{ℓ} {A : Type ℓ} → (A → A) → Type _
+isInvolution f = ∀ x → f (f x) ≡ x
+
+module _ {ℓ} {A : Type ℓ} {f : A → A} (invol : isInvolution f) where
+  open Iso
+
+  involIso : Iso A A
+  involIso .fun = f
+  involIso .inv = f
+  involIso .rightInv = invol
+  involIso .leftInv = invol
+
+  involIsEquiv : isEquiv f
+  involIsEquiv = isoToIsEquiv involIso
+
+  involEquiv : A ≃ A
+  involEquiv = f , involIsEquiv
+
+  involPath : A ≡ A
+  involPath = ua involEquiv
+
+  involEquivComp : compEquiv involEquiv involEquiv ≡ idEquiv A
+  involEquivComp
+    = equivEq (compEquiv involEquiv involEquiv) (idEquiv A) (λ i x → invol x i)
```

Comment:
Could the first two arguments of `equivEq` be implicit? I'm kind of puzzled why they're not, but maybe Agda needs them not to be?

Reply:
Apparently they can. I'm not sure why I thought they couldn't.

Reply:
Maybe `equivEq` should be redefined so that they're implicit?

Reply:
Possibly. I'm not sure if it's used often in situations where the result type is under-determined.

Reply:
`grep` tells me that it's almost exclusively used with `_` so let's change it, but in a dedicated PR. I'll open an issue about it.